src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -346,6 +358,18 @@ public bool ResolveDocumentPath(ref string document)
 
         #region Public Class Operational Methods
 
+        /// <summary>
+        /// Assigns an IPathResolver on demand with the same behavior as the Ctor.
+        /// </summary>
+        /// <param name="resolver"></param>
+        internal void AssignIPathResolver(IPathResolver resolver)
+        {
+            hostApplicationDirectory = string.Empty;
```

Comment:
`hostApplicationDirectory` is set from Path Manager constructor right? Why do we need to set it again here?